docs/source/reference/storage.rst

```diff
@@ -1,303 +1,291 @@
+    :code:`close()` on a file will upload the entire file to the bucket.
     Any subsequent reads, either using SkyPilot Storage or external utilities (such as
     aws/gsutil cli) will see the latest data.
 
 .. note::
-    :code:`MOUNT` mode does not support the full POSIX interface and some file
-    operations may fail. Most notably, random writes and append operations are
-    not supported.
+    SkyPilot does not guarantee preservation of file permissions when attaching
+    buckets. You may need to set file permissions during task execution.
 
 .. note::
-    Storage only supports uploading directories (i.e., :code:`source` cannot be a file).
-    To upload a single file to a bucket, please put in a directory and specify the directory as the source.
-    To directly copy a file to a VM, please use regular :ref:`file mounts <file-mounts-example>`.
+    Symbolic links are handled differently in :code:`file_mounts` depending on whether buckets are used.
+    For bucket mounts, symbolic links are not copied to remote.
+    For local ``file_mounts`` that are directly rsynced to the VM,
+    the symbolic links are directly copied, not their target data.
+    The targets must be separately mounted or else the symlinks may break.
+
+Common patterns
+---------------
 
-Creating a shared file system
------------------------------
+Storing and sharing outputs of tasks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-SkyPilot Storage can also be used to create a shared file-system backed by a remote object store (e.g., S3)
-that multiple tasks on different nodes can read and write to. This allows developers to pass files
-between workers and even use files as a medium for inter-process communication (IPC).
+You can directly write the outputs of your tasks to a cloud bucket by creating
+a new bucket and using it in :code:`MOUNT` mode.
 
-To create a shared filesystem, simply create a Storage object without a source
-and use mount mode when attaching it to your tasks like so:
+**💡 Example use-case**: Writing model checkpoints, logs from training runs.
 
 .. code-block:: yaml
 
-    file_mounts:
-      /sharedfs:
-        name: my-sky-sharedfs
-        mode: MOUNT
+  # Creates an empty bucket. Any writes to /my_data will be replicated to s3://my-sky-bucket
+  file_mounts:
+    /my_data:
+        name: my-sky-bucket
+        store: s3
+
+Your task can then write files to :code:`/my_data` and they will be automatically
+be uploaded to the cloud.
 
 
-Here is a `simple example <https://github.com/skypilot-org/skypilot/blob/master/examples/storage/pingpong.yaml>`_
-using SkyPilot Storage to perform communication between processes using files.
+Avoid re-uploading data on every run
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
+Compared directly using local paths in file_mounts, uploading to a bucket can
+be faster since it is persistent and requires fewer uploads from your local
+machine if the data is not changing.
```

Comment:
```suggestion
be faster since it is persistent.
```
Even if data is changing, this format is supported, right? (Like the note suggests below)

Reply:
Also, what's the reason this is faster vs. uploading to VMs? Is it persistent, or object storage having larger bandwidths than VMs.